homeassistant/components/sensor/__init__.py

```diff
@@ -270,6 +271,7 @@ def unit_of_measurement(self) -> str | None:
     def state(self) -> Any:
         """Return the state of the sensor and perform unit conversions, if needed."""
         unit_of_measurement = self.native_unit_of_measurement
+        device_class = self.device_class
```

Comment:
The device_class should be something static, and not potentially change the state is updated.

Reply:
just saw you use it as a local variable, that is different.